server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -715,16 +710,10 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
      */
     void writeIncompatibleSnapshots(RepositoryData repositoryData) throws IOException {
         assert isReadOnly() == false; // can not write to a read only repository
-        final BytesReference bytes;
-        try (BytesStreamOutput bStream = new BytesStreamOutput()) {
-            try (StreamOutput stream = new OutputStreamStreamOutput(bStream);
-                 XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON, stream)) {
-                repositoryData.incompatibleSnapshotsToXContent(builder);
-            }
-            bytes = bStream.bytes();
-        }
+        final XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
```

Comment:
Inline builder as it's done with `repositoryData.snapshotsToXContent`?

Reply:
Can't do it here since the API is a little different here :( the call to `org.elasticsearch.repositories.RepositoryData#incompatibleSnapshotsToXContent` returns `void`.

Reply:
I adjusted this now after all to make the API a little more fluent :)